tests/scripts/thread-cert/border_router/test_advertising_proxy.py

```diff
@@ -176,7 +176,34 @@ def test(self):
         self.host_check_mdns_service(host, '2001::2', 'my-service-1')
 
         #
-        # 7. Check if the expired service is removed by the Advertising Proxy.
+        # 7. Remove a single service and verify that the remaining one can still
+        #    be discovered.
+        #
+
+        client.srp_client_remove_service('my-service-1', '_ipps._tcp')
+        self.simulator.go(5)
+
+        self.check_host_and_service(server, client, '2001::2', 'my-service')
+        self.host_check_mdns_service(host, '2001::2', 'my-service')
+        self.assertIsNone(host.discover_mdns_service('my-service-1', '_ipps._tcp', 'my-host'))
+
+        #
+        # 8. Update both the host and the service in a loop and make sure the
+        #    Advertising Proxy can follow.
+        #
+
+        for host_address_suffix, service_port in ((1, 12341), (2, 12342), (3, 12343), (2, 12345)):
+            host_address = f'2001::{host_address_suffix}'
+            client.srp_client_set_host_address(host_address)
+            client.srp_client_clear_service('my-service', '_ipps._tcp')
+            client.srp_client_add_service('my-service', '_ipps._tcp', service_port)
```

Comment:
If I understand it correctly, this test case is trying to include `Host` and `Service` updates in a single SRP message. I know that the SRP client will wait for `OPENTHREAD_CONFIG_SRP_CLIENT_UPDATE_TX_DELAY` (10ms) before sending a SRP message when `Host` or `Service` info is updated to collect multiple updates in a single SRP message. Considering that the commands in the test script will be mapped to separate CLI commands, there is no guarantee that `client.srp_client_set_host_address` and `client.srp_client_add_service` can be executed in `10ms`.

Suggest changing to
```c++
client.srp_client_clear_service('my-service', '_ipps._tcp')
client.srp_client_set_host_address(host_address)
client.srp_client_add_service('my-service', '_ipps._tcp', service_port)
```

Reply:
I think it's a valid point. I'll change it as you suggested.